kamu-adapter-graphql/src/scalars/dataset.rs

```diff
@@ -252,7 +260,7 @@ pub enum DataQueryResult {
 }
 
 impl DataQueryResult {
-    pub fn success(schema: DataSchema, data: DataBatch, limit: u64) -> DataQueryResult {
+    pub fn success(schema: Option<DataSchema>, data: DataBatch, limit: u64) -> DataQueryResult {
```

Comment:
Should we also introduce an `empty()` factory function, so that `success()` continues to take value not `Option`? This way it won't be possible to create a result with data but without a schema.